### PR TITLE
Feature/label description

### LINKED ISSUE #2114: Label: add a description
**Why?** 

By providing a clear description we could feed an unsuppervised front in order to run the synchronization.

**What?**

Add the following field into the `Label` entity:

 - **description** (String) optional 

Steps:
 - Modify the storage API, add relevant test case in storage contract and adapt Cassandra and Postgres
 - Modify the JMAP layer to allow do get and edit the description and add the relevant integration tests

### PR DESCRIPTION
**Pull request by AlaeMghirbi**
Add optional `description` field (String) to the Label entity to provide clear descriptions for frontend synchronization.

closes #2114

## Summary by CodeRabbit

* **New Features**
  * Labels now include an optional description field on create, update, and in query responses; description can be set, cleared (null) and returned.

* **Documentation**
  * API docs and upgrade instructions updated with schema migration steps and pre-deployment guidance to add the description column.

* **Tests**
  * Unit and integration tests updated to cover creation, update, retrieval and edge cases (present, null, missing) for label descriptions.

### CODE REVIEW

File: tmail-backend/jmap/extensions-postgres/src/main/java/com/linagora/tmail/james/jmap/label/PostgresLabelDAO.java [hunk 1]
```diff
@@ -114,6 +120,17 @@ private Optional<UpdateSetMoreStep<Record>> addUpdateColor(Option<Color> newColo
                 .orElseGet(() -> originalUpdateStatement.set(COLOR, color.value())));
     }
 
+    private Optional<UpdateSetMoreStep<Record>> addUpdateDescription(Option<DescriptionUpdate> newDescription, UpdateSetFirstStep<Record> originalUpdateStatement, Optional<UpdateSetMoreStep<Record>> updateDisplayNameStatement, Optional<UpdateSetMoreStep<Record>> updateColorStatement) {
+        return OptionConverters.toJava(newDescription)
+            .map(descUpdate -> OptionConverters.toJava(descUpdate.value())
+                .map(description -> updateColorStatement.map(statement -> statement.set(DESCRIPTION, description))
+                    .orElseGet(() -> updateDisplayNameStatement.map(statement -> statement.set(DESCRIPTION, description))
+                        .orElseGet(() -> originalUpdateStatement.set(DESCRIPTION, description))))
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions-postgres/src/main/java/com/linagora/tmail/james/jmap/label/PostgresLabelDAO.java]: Too much indent levels
